index.html

```diff
@@ -102,29 +109,79 @@ <h2>Items in cart</h2>
       function initializeCart() {
         const itemsInCart = []
 
+        const cartListElement = document.getElementById('cart-list')
+
+        function addToCart(array, item) {
+          array.push({
+          product: item.product,
+          price: item.price,
+          count: 1,
+          })
+        }
+
         Array.from(document.querySelectorAll('.add-to-cart')).forEach(
           (button) => {
             button.addEventListener('click', () => {
               console.log('clicked add to cart')
+              
+              const typesOfItemsInCart = itemsInCart.length
 
-              itemsInCart.push({
+              const latestCartItem = {
                 product: button.dataset.product,
                 price: Number(button.dataset.price),
-              })
-
-              const itemsInCartCount = itemsInCart.length
+              }
+
+              // This will keep track if the prodcut has been added to itemsInCart
+              let isInCart = false
+
+              // Check if the cart is empty
+              if (itemsInCart.length === 0) {
+                addToCart(itemsInCart, latestCartItem)
+                isInCart = true
+              } else {
+                // Cart isn't empty, so is the same type of product already exist?
+                for (let i=0; i < typesOfItemsInCart; i++) {
+                  if (itemsInCart[i].product === latestCartItem.product) {
+                    itemsInCart[i].count++
+                    isInCart = true
+                    break
+                  }
+                }
+              }
+ 
+              // Adding new type of product to cart
+              if (isInCart === false) {
+                addToCart(itemsInCart, latestCartItem)
+                isInCart = true
+              }
```

Comment:
**Performance improvement suggestion** pt 2: Expand the logic here: if an item is in the cart, update the existing HTML. If an item is not in the cart, create the new HTML--essentially what you were doing on lines 181-183. You can use the `latestCartItem` to create or update the HTML, now that we have an updated count as a property of that object.

```suggestion
              if (isInCart) {
                // Update cart view with existing product, by finding the
                // existing list item by ID
                const cartListItemElement = document.getElementById(productId)

                cartListItemElement.innerText = `${latestCartItem.product} ($${
                  latestCartItem.price
                } / per) x ${latestCartItem.count} = ${(
                  latestCartItem.price * latestCartItem.count
                ).toFixed(2)}`
                // cartListElement.appendChild(cartListItemElement)
              } else {
                // Set initial count
                latestCartItem.count = 1

                addToCart(itemsInCart, latestCartItem)
                isInCart = true

                // Update cart view with new product
                const cartListItemElement = document.createElement('li')

                // We need to set an ID (or class) on the newly created list item
                // so that we can find it later to update the existing list item
                cartListItemElement.id = productId

                cartListItemElement.innerText = `${latestCartItem.product} ($${
                  latestCartItem.price
                } / per) x ${latestCartItem.count} = ${(
                  latestCartItem.price * latestCartItem.count
                ).toFixed(2)}`
                cartListElement.appendChild(cartListItemElement)
              }
```